Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Extensions/ApiGatewayResponseExtensions.cs

```diff
@@ -0,0 +1,108 @@
+﻿namespace Amazon.Lambda.TestTool.Extensions;
+
+using Amazon.Lambda.APIGatewayEvents;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using System;
+using System.IO;
+using System.Text;
+
+/// <summary>
+/// Provides extension methods for converting API Gateway responses to HttpResponse objects.
+/// </summary>
+public static class ApiGatewayResponseExtensions
+{
+    /// <summary>
+    /// Converts an APIGatewayProxyResponse to an HttpResponse.
+    /// </summary>
+    /// <param name="apiResponse">The API Gateway proxy response to convert.</param>
+    /// <returns>An HttpResponse object representing the API Gateway response.</returns>
+    public static HttpResponse ToHttpResponse(this APIGatewayProxyResponse apiResponse)
+    {
+        var httpContext = new DefaultHttpContext();
+        var response = httpContext.Response;
+
+        response.StatusCode = apiResponse.StatusCode;
+
+        SetResponseHeaders(response, apiResponse.Headers, apiResponse.MultiValueHeaders);
```

Comment:
i updated to set content length. for the default content-type, theres actually a couple of cases for how that can get set. ive written test cases for those and it may be easier to read them than me typing :D

Reply:
so i dug more into this. this was tripping me up for a couple of hours. the default content type actually depends on what kind of api gateway is being used. If content-type is not specified and a REST api is being used, the content type is application/.json. but if its an http api being used (for v1) its text/plain. but then a http (v2) can sometimes infer the application type is json...